File: mobile/app/src/main/java/com/bevietnam/core/model/Place.kt
package com.bevietnam.core.model

/**
 * Mô hình dữ liệu đại diện cho một địa điểm/danh lam thắng cảnh văn hóa (Place/Point of Interest).
 *
 * Khớp với Backend schema `PlaceSchema` tại `GET /api/v1/places`.
 *
 * @property id Định danh duy nhất của địa điểm.
 * @property name Tên của địa điểm.
 * @property category Danh mục phân loại địa điểm (ví dụ: "temple", "museum", "park", "district").
 * @property description Mô tả chi tiết hoặc giới thiệu về giá trị văn hóa, ý nghĩa của địa điểm.
 * @property latitude Vĩ độ tọa độ GPS của địa điểm.
 * @property longitude Kinh độ tọa độ GPS của địa điểm.
 * @property imageUrl Đường dẫn URL dẫn tới hình ảnh đại diện của địa điểm (có thể null).
 * @property referenceUrl Đường dẫn URL tham chiếu bên ngoài về địa điểm (có thể null).
 */
data class Place(
    val id: String,
    val name: String,
    val category: String,
    val description: String,
    val latitude: Double,
    val longitude: Double,
    val imageUrl: String? = null,
    val referenceUrl: String? = null
)

File: backend/app/schemas/place.py
"""Place schemas — API contract for place-related endpoints."""
from pydantic import BaseModel, Field
from typing import Optional, List


class PlaceCreateRequest(BaseModel):
    place_id: Optional[str] = Field(None, alias="place_id")
    name: str
    category: str
    description: str
    latitude: float
    longitude: float
    image_url: Optional[str] = None
    reference_url: Optional[str] = None

    model_config = {
        "populate_by_name": True,
    }


class PlaceSchema(BaseModel):
    id: str
    name: str
    category: str  # "temple" | "museum" | "park" | "district" | ...
    description: str
    latitude: float
    longitude: float
    image_url: Optional[str] = None
    reference_url: Optional[str] = None

    model_config = {
        "from_attributes": True,
    }


class PlaceImportRequest(BaseModel):
    file_path: Optional[str] = None
    items: List[PlaceCreateRequest] = Field(default_factory=list)

    model_config = {
        "populate_by_name": True,
    }


class PlacesResponse(BaseModel):
    total: int
    items: List[PlaceSchema]
